src/test/coinscachepair_tests.cpp

```diff
@@ -0,0 +1,216 @@
+// Copyright (c) 2024-present The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <coins.h>
+
+#include <boost/test/unit_test.hpp>
+
+#include <list>
+
+BOOST_AUTO_TEST_SUITE(coinscachepair_tests)
+
+static constexpr auto NUM_NODES{4};
+
+std::list<CoinsCachePair> CreatePairs(CoinsCachePair& sentinel)
+{
+    std::list<CoinsCachePair> nodes;
+    for (auto i{0}; i < NUM_NODES; ++i) {
+        nodes.emplace_back();
+
+        auto node{std::prev(nodes.end())};
+        node->second.AddFlags(CCoinsCacheEntry::DIRTY, *node, sentinel);
```

Comment:
`->first` and `->second` seems really foreign throughout the usages, if we could replace the `std::pair<const COutPoint, CCoinsCacheEntry>` with a 
```C++
class CoinsCachePair {
public:
    const COutPoint outpoint;
    CCoinsCacheEntry entry;
    
    CoinsCachePair() : outpoint(), entry() {
        entry.SelfRef(*this);
    }
    ...
```

we could have `sentinel.entry.m_prev` instead of `sentinel.second.m_prev`, which is more explanatory, and we wouldn't need to repeat every init as:
```C++
    CoinsCachePair sentinel{};
    sentinel.second.SelfRef(sentinel);
```

since we can now do that in the constructor. What do you think? I understand if you want to do it in a separate PR as well, but I will gladly reapprove as well.

Reply:
Ah, thanks, makes sense